e2e/test/scenarios/question/notebook-data-source.cy.spec.ts

```diff
@@ -0,0 +1,357 @@
+      openReviewsTable();
+      openNotebook();
+      cy.findByTestId("data-step-cell").should("have.text", "Reviews").click();
+      popover().within(() => {
+        cy.findByTestId("source-database").should(
+          "have.text",
+          "Sample Database",
+        );
+        cy.findByLabelText("Reviews").should(
+          "have.attr",
+          "aria-selected",
+          "true",
+        );
+      });
+    });
+
+    it("should correctly display the source data for a simple saved question", () => {
+      visitQuestion(ORDERS_COUNT_QUESTION_ID);
+      openNotebook();
+      cy.findByTestId("data-step-cell").should("have.text", "Orders").click();
+      popover().within(() => {
+        cy.findByTestId("source-database").should(
+          "have.text",
+          "Sample Database",
+        );
+        cy.findByLabelText("Orders").should(
+          "have.attr",
+          "aria-selected",
+          "true",
+        );
+      });
+    });
+
+    it(
+      "should correctly display a table from a multi-schema database (metabase#39807)",
+      { tags: "@external" },
+      () => {
+        const dialect = "postgres";
+        const TEST_TABLE = "multi_schema";
+
+        const dbName = "Writable Postgres12";
+        const schemaName = "Wild";
+        const tableName = "Animals";
+
+        resetTestTable({ type: dialect, table: TEST_TABLE });
+        restore(`${dialect}-writable`);
+
+        cy.signInAsAdmin();
+
+        resyncDatabase({
+          dbId: WRITABLE_DB_ID,
+        });
+
+        startNewQuestion();
+        popover().within(() => {
+          cy.findByText("Raw Data").click();
+          cy.findByText(dbName).click();
+          cy.findByText(schemaName).click();
+          cy.findByText(tableName).click();
+        });
+        visualize();
+        saveQuestion("Beasts");
+
+        openNotebook();
+        cy.findByTestId("data-step-cell").should("contain", tableName).click();
+        popover().within(() => {
+          cy.findByTestId("source-database").should("have.text", dbName);
+          cy.findByTestId("source-schema").should("have.text", schemaName);
+        });
+      },
+    );
+
+    it("should correctly display a table as the model's source when editing simple model's query", () => {
+      cy.visit(`/model/${ORDERS_MODEL_ID}/query`);
+
+      cy.findByTestId("data-step-cell").should("have.text", "Orders").click();
+      popover().within(() => {
+        cy.findByTestId("source-database").should(
+          "have.text",
+          "Sample Database",
+        );
+        cy.findByLabelText("Orders").should(
+          "have.attr",
+          "aria-selected",
+          "true",
+        );
+      });
+    });
+  });
+
+  describe("saved entity as a source (aka the virtual table)", () => {
+    const modelDetails = {
+      name: "GUI Model",
+      query: { "source-table": REVIEWS_ID, limit: 1 },
+      display: "table",
+      type: "model" as const,
```

Comment:
This was needed because otherwise TS complains that "string" is not compatible with the type `CardType | undefined`